src/Type/WPObjectType.php

```diff
@@ -140,6 +142,16 @@ public static function prepare_fields( $fields, $type_name ) {
 			 */
 			$fields = apply_filters( "graphql_{$uc_type_name}_fields", $fields );
 
+			/**
+			 * If the object defines input fields, apply a centralized filter for
+			 * input fields. This was previously handled by WPInputObjectType, but
+			 * the fields definition is now trapped in a closure so all filters need
+			 * to happen in one place.
+			 */
+			if ( isset( $config['kind'] ) && 'input' === $config['kind'] ) {
+				$fields = WPInputObjectType::prepare_fields( $fields, $type_name, $config );
+			}
```

Comment:
@chriszarate in what situation does `kind === 'input'` and evoke a `WPObjectType`? 

if you `register_graphql_type()` and specify `'kind' => 'input'` or `'register_graphql_input_type()'` where `'kind' => 'input'` is set for you, then `WPObjectType` shouldn't even be getting instantiated, but `WPInputObjectType` should be, as seen here: 

https://github.com/wp-graphql/wp-graphql/blob/develop/src/TypeRegistry.php#L378-L380

I could easily be missing something though. . .can you clarify how you get to this point where `kind === input` in a WPObjectType?

Reply:
Updated to centralize `prepare_fields` calls per our conversation.